plugins/teststeps/awsfileupload/awsFileUpload.go

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package awsfileupload
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/facebookincubator/contest/pkg/event"
+	"github.com/facebookincubator/contest/pkg/event/testevent"
+	"github.com/facebookincubator/contest/pkg/target"
+	"github.com/facebookincubator/contest/pkg/test"
+	"github.com/facebookincubator/contest/pkg/xcontext"
+	"github.com/facebookincubator/contest/plugins/teststeps"
+)
+
+// Name is the name used to look this plugin up.
+var Name = "awsFileUpload"
+
+// Event names for this plugin.
+const (
+	EventURLtoReport = event.Name("URLtoReport")
+)
+
+// Events defines the events that a TestStep is allow to emit
+var Events = []event.Name{
+	EventURLtoReport,
+}
+
+// FileUpload is used to retrieve all the parameter, the plugin needs.
+type FileUpload struct {
+	path            *test.Param // Path to file that shall be uploaded
+	filename        *test.Param // Filename to file that shall be uploaded
+	s3region        string      // AWS server region that shall be used
```

Comment:
is this field mandatory? If not can you add it to the docstring explaining what's the default otherwise?

Reply:
Unless there is no config file that defines the region it is mandatory. There is no default otherwise.